pallets/xcm-weight-trader/src/lib.rs

```diff
@@ -265,37 +265,37 @@ pub mod pallet {
 		pub fn query_acceptable_payment_assets(
 			xcm_version: xcm::Version,
 		) -> Result<Vec<VersionedAssetId>, XcmPaymentApiError> {
-			if !matches!(xcm_version, 3 | 4) {
+			if !matches!(xcm_version, 3 | 5) {
 				return Err(XcmPaymentApiError::UnhandledXcmVersion);
 			}
 
-			let v4_assets = [VersionedAssetId::V4(XcmAssetId::from(
+			let v5_assets = [VersionedAssetId::from(XcmAssetId::from(
 				T::NativeLocation::get(),
 			))]
 			.into_iter()
 			.chain(
 				SupportedAssets::<T>::iter().filter_map(|(asset_location, (enabled, _))| {
-					enabled.then(|| VersionedAssetId::V4(XcmAssetId(asset_location)))
+					enabled.then(|| VersionedAssetId::from(XcmAssetId(asset_location)))
 				}),
 			)
 			.collect::<Vec<_>>();
 
 			if xcm_version == 3 {
```

Comment:
I know that this was already there, but we should change it to:
```suggestion
			if xcm_version == xcm::v3::VERSION {
```

Reply:
Ideally, we can remove `v3` and `v4` once all existing locations have been migrated.

Reply:
Addressed in https://github.com/moonbeam-foundation/moonbeam/pull/3224/commits/936b6c3de710cc39ffbfd67a35c8d15523b31e67